### PR TITLE
fix: Use the ILambdaSerializer to serialize the response body when returning an IHttpResult and generate documentation comments for public members

### PR DESCRIPTION
**Pull request by 96malhar**
*Issue #, if available:*
* https://github.com/aws/aws-lambda-dotnet/issues/1727 (**DOTNET-7489**)
* https://github.com/aws/aws-lambda-dotnet/issues/1715 (**DOTNET-7459**)
* https://github.com/aws/aws-lambda-dotnet/discussions/1733

*Description of changes:*
* chore: Generate documentation comments for public memebers and add the \<auto-generated/\> hint to the generated files so that they are automatically skipped by the analyzers.
* fix: Use the `ILambdaSerializer` to serialize the response body when returning an `IHttpResult`
    * **BREAKING CHANGE**: Removed the `JsonContext` property in the `HttpResultSerializationOptions` class and replaced it with an `ILambdaSerializer`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Libraries/test/TestExecutableServerlessApp/SourceGenerationSerializationExample.cs [hunk 1]
```diff
@@ -22,8 +22,6 @@ public IHttpResult GetPerson(ILambdaContext context)
         }
     }
 
-    [JsonSerializable(typeof(APIGatewayHttpApiV2ProxyRequest))]
-    [JsonSerializable(typeof(APIGatewayHttpApiV2ProxyResponse))]
```
Comment (96malhar) [inline Libraries/test/TestExecutableServerlessApp/SourceGenerationSerializationExample.cs:26]: Since the lambda function returns an `IHttpResult`, it does not need to serialize these types.

**Review by normj — APPROVED** [review]
You could also add the version bump as well to the PR but the changes look good.

File: Libraries/test/Amazon.Lambda.Annotations.SourceGenerators.Tests/SourceGeneratorTests.cs [hunk 2]
```diff
@@ -284,7 +284,6 @@ public async Task VerifyFunctionInSubNamespace()
         {
             var expectedTemplateContent = (await File.ReadAllTextAsync(Path.Combine("Snapshots", "ServerlessTemplates", "subnamespace.template"))).ToEnvironmentLineEndings();
             var expectedSubNamespaceGenerated = (await File.ReadAllTextAsync(Path.Combine("Snapshots", "Functions_ToUpper_Generated.g.cs"))).ToEnvironmentLineEndings();
-            var expectedProgramGenerated = (await File.ReadAllTextAsync(Path.Combine("Snapshots", "Program.g.cs"))).ToEnvironmentLineEndings();
```
Comment (ashovlin) [inline Libraries/test/Amazon.Lambda.Annotations.SourceGenerators.Tests/SourceGeneratorTests.cs:287]: Why remove this (and the next one)? Were we missing the assert against it?

File: Libraries/test/Amazon.Lambda.Annotations.SourceGenerators.Tests/Snapshots/FunctionsZipOutput_ToLower_Generated.g.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using Microsoft.Extensions.DependencyInjection;
+using Amazon.Lambda.Core;
+
+namespace TestServerlessApp.Sub1
+{
+    public class FunctionsZipOutput_ToLower_Generated
```
Comment (ashovlin) [inline Libraries/test/Amazon.Lambda.Annotations.SourceGenerators.Tests/Snapshots/FunctionsZipOutput_ToLower_Generated.g.cs:14]: Why are these files being added?
1. `FunctionsZipOutput_ToLower_Generated`
2. `ParameterlessMethodWithResponse_NoParameterWithResponse_Generated`
3. `ParameterlessMethods_NoParameter_Generated`


They seem unrelated to the new `OkResponseWithCustomSerializer`, so just trying to understand how/why they're here now.

Comment (96malhar) [reply Libraries/test/Amazon.Lambda.Annotations.SourceGenerators.Tests/Snapshots/FunctionsZipOutput_ToLower_Generated.g.cs]: I had updated the snapshots programmatically. These may have been generated as a side effect. Deleted these files.

File: Libraries/test/Amazon.Lambda.Annotations.SourceGenerators.Tests/SourceGeneratorTests.cs (hunk 2 above)
Comment (96malhar) [reply Libraries/test/Amazon.Lambda.Annotations.SourceGenerators.Tests/SourceGeneratorTests.cs:287]: The `Program.g.cs` does not even get generated since the source files being supplied to  the test were from the `TestServerlessApp` and not the `TestExecutableServerlessApp`. 

We weren't asserting against `expectedProgramGenerated` and hence removed it from both places.

File: Libraries/src/Amazon.Lambda.Annotations/APIGateway/HttpResults.cs [hunk 4]
```diff
@@ -356,7 +353,7 @@ public static IHttpResult NewResult(HttpStatusCode statusCode, object body = nul
             Justification = "When using this library with the Native AOT the SourceGeneratorLambdaJsonSerializer has to be registered which will provide the information needed for JsonSerializerContext and avoid the reflection based JsonSerializer.Serialize call")]
```
[the remark this answers is not in the document]
Comment (96malhar) [reply Libraries/src/Amazon.Lambda.Annotations/APIGateway/HttpResults.cs]: Removed the suppression messages.

**Review by ashovlin — APPROVED** [review]
